### ISSUE TITLE
MediaTexts and Anchors get lost - liquid templates

### ISSUE THREAD

**Issue by bashuss** [OrchardCMS/OrchardCore#15425]
### Describe the bug
using OC 1.7
When accessing the media field, which is part of my BlogPost ContentType by `Model.Content.BlogPost-Carousel`
and passing it to new shape as parameter, Anchors and MediaTexts get lost. Paths are still available.

this creates the problem:
`{{ "CarouselShape" | shape_new: Media: Model.Content.BlogPost-Carousel | shape_render }}`

I get in CarouselShape:
`{{ Model.Media.Paths | json }}` => Array of paths
`{{ Model.Media.MediaTexts | json }}` => null
`{{ Model.Media.Anchors | json }}` => null

Calling in this way works for me:
`{{ "CarouselShape" | shape_new: Media: Model.ContentItem.Content.BlogPost.Carousel | shape_render }}`

### To Reproduce
1. Create or edit a ContentType "Test", with media field called "Carousel", which allows Anchors and MediaTexts
2. Create a Template for ContentType "Test", named "Content__Test", which does something like this:
```
<div>
<h2>this fails: Model.Content.Test-Carousel</h2>
{{ "TestShape" | shape_new: Media: Model.Content.Test-Carousel | shape_render }}
</div>
<div>
<h2>this works: Model.ContentItem.Content.Test.Carousel</h2>
{{ "TestShape" | shape_new: Media: Model.ContentItem.Content.Test.Carousel | shape_render }}
</div>
```
3. Create a Template for Shape "TestShape", which does something like this:
```
<ul>
<li>Paths:  {{ Model.Media.Paths | json }}</li>
<li>MediaTexts: {{ Model.Media.MediaTexts | json }}</li>
<li>Anchors: {{ Model.Media.Anchors | json }}</li>
</ul>
```
4. Create a ContentItem for Test ContentType with images, MediaTexts and Anchors
5. Preview
6. The result show different behavior for both calls, as described above.

### Expected behavior
I would expect, when Paths are available, that also MediaTexts and Anchors are available in the same way.

**Comment by Skrypt** [comment]
Have you tried this?

<li>MediaTexts: {{ Model.Media.MediaTexts }}</li>
<li>Anchors: {{ Model.Media.Anchors }}</li>

**Comment by hishamco** [comment]
Did you try it in OC `1.8.x`?

**Comment by bashuss** [comment]
> Have you tried this?
> 
> ```
> <li>MediaTexts: {{ Model.Media.MediaTexts }}</li>
> <li>Anchors: {{ Model.Media.Anchors }}</li>
> ```

No, but I tried this:
```
{% if Model.Media.MediaTexts %}
{% assign text = Model.Media.MediaTexts[index] %}
{% endif %}
```
Text was never set until I used the workaround described above.

**Comment by bashuss** [comment]
> Did you try it in OC `1.8.x`?

No, still have problems with migrating my main menu handling in 1.8 - cannot get my front end to work properly. Just a matter of time, which I don't have enough right now. Will get back to you, when I had a chance to migrate.

**Comment by gvkries** [comment]
This is the expected behavior. When accessing `Model.Content.BlogPost-Carousel`, you're passing the shape of the media field to the new shape. As a result, the field values become available via `Model.Media.Field.Content.Paths` and `Model.Media.Field.Content.MediaTexts` etc. in the new shape.

However, your second approach is preferable, as it passes only the media field itself to the new shape.
